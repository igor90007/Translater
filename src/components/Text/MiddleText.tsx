import React from 'react'
import EStyleSheet from 'react-native-extended-stylesheet'

import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

export interface IMiddleTextProps extends TextProps {
  additionalStyles?: StyleProp<TextStyle>
}

export const MiddleText: React.FC<IMiddleTextProps> = ({ children, additionalStyles }) => (
  <Text style={[styles.text, additionalStyles]}>{children}</Text>
)

const styles = EStyleSheet.create({
  text: {
    color: '$textColor',
    fontSize: '$textDefault',
  },
})
