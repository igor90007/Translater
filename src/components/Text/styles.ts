import EStyleSheet from 'react-native-extended-stylesheet'

import { heightPercentageToDP as hp } from 'react-native-responsive-screen'

export default EStyleSheet.create({
  text: {
    color: '$textColor',
    fontSize: '$textDefault',
  },
})
