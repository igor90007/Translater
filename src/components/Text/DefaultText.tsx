import React from 'react'

import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

import styles from './styles'

export interface ITextProps extends TextProps {
  children?: string
  additionalStyles?: StyleProp<TextStyle>
}

const DefaultText: React.FC<ITextProps> = ({ children, additionalStyles }) => (
  <Text style={[styles.text, additionalStyles]}>{children}</Text>
)

export default DefaultText
