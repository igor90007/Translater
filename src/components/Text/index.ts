import DefaultText from './DefaultText'

import styles from './styles'

export { DefaultText, styles }
