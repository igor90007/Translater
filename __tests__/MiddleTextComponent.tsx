import React from 'react'
import 'react-native'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

import { MiddleText } from '../src/components/Text'

test('renders correctly MiddleText component', () => {
  const tree = renderer.create(<MiddleText additionalStyles={{ color: 'red' }}>test</MiddleText>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('MiddleText component with null properties', () => {
  const tree = renderer.create(<MiddleText />).toJSON()
  expect(tree).toMatchSnapshot()
})
