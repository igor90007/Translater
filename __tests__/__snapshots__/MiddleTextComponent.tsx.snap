// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiddleText component with null properties 1`] = `
<Text
  style={
    Array [
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`renders correctly MiddleText component 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "color": "red",
      },
    ]
  }
>
  test
</Text>
`;
